From 78a7fac21c2c1237d738e71432b36cf69209db0e Mon Sep 17 00:00:00 2001
From: Flavio Cruz <flaviocruz@gmail.com>
Date: Sat, 10 Dec 2022 22:55:35 -0500
Subject: [PATCH] Add support for x86_64-*-gnu-* targets

---
 bfd/config.bfd   | 5 +++++
 ld/configure.tgt | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/bfd/config.bfd b/bfd/config.bfd
index 0bc27fdce9..3a1c9f716e 100644
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -701,6 +701,11 @@ case "${targ}" in
     targ_selvecs="i386_elf32_fbsd_vec iamcu_elf32_vec i386_coff_vec i386_pei_vec x86_64_pe_vec x86_64_pei_vec i386_elf32_vec x86_64_elf64_vec"
     want64=true
     ;;
+  x86_64-*-gnu*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs="i386_elf32_vec iamcu_elf32_vec x86_64_elf32_vec"
+    want64=true
+    ;;
   x86_64-*-haiku*)
     targ_defvec=x86_64_elf64_vec
     targ_selvecs="i386_elf32_vec"
diff --git a/ld/configure.tgt b/ld/configure.tgt
index 741b246f67..d5f1278800 100644
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -995,6 +995,11 @@ x86_64-*-rdos*)		targ_emul=elf64rdos
 			;;
 x86_64-*-cloudabi*)	targ_emul=elf_x86_64_cloudabi
 			;;
+x86_64-*-gnu*)	targ_emul=elf_x86_64
+			targ_extra_emuls="elf32_x86_64 elf_iamcu elf_i386"
+			targ_extra_libpath="elf_i386 elf32_x86_64"
+			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
+			;;
 x86_64-*-haiku*)	targ_emul=elf_x86_64_haiku
 			targ_extra_emuls="elf_x86_64 elf_i386_haiku"
 			;;
-- 
2.37.2

