---
 linux/dev/glue/block.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/linux/dev/glue/block.c
+++ b/linux/dev/glue/block.c
@@ -207,7 +207,10 @@ int
 blk_dev_init ()
 {
 #ifdef CONFIG_BLK_DEV_IDE
-  ide_init ();
+  extern char *kernel_cmdline;
+  if (strncmp(kernel_cmdline, "noide", 5) &&
+      !strstr(kernel_cmdline, " noide"))
+    ide_init ();
 #endif
 #ifdef CONFIG_BLK_DEV_FD
   floppy_init ();
