---
 linux/src/drivers/block/ide.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/linux/src/drivers/block/ide.c
+++ b/linux/src/drivers/block/ide.c
@@ -3725,7 +3725,10 @@ static void probe_for_hwifs (void)
 #ifdef CONFIG_BLK_DEV_PROMISE
 	init_dc4030();
 #endif
-	ahci_probe_pci();
+	extern char *kernel_cmdline;
+	if (strncmp(kernel_cmdline, "noahci", 6) &&
+	    !strstr(kernel_cmdline, " noahci"))
+		ahci_probe_pci();
 }
 
 static int hwif_init (int h)
