--- coreutils-8.24/Makefile.in	2015-07-03 23:23:13.000000000 +0200
+++ src/coreutils-8.24/Makefile.in	2015-12-27 01:21:48.151688848 +0100
@@ -2308,7 +2308,7 @@
   }
 man1dir = $(mandir)/man1
 NROFF = nroff
-MANS = $(man1_MANS)
+MANS =
 HEADERS = $(noinst_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
@@ -3853,7 +3853,7 @@
 lispdir = @lispdir@
 localedir = @localedir@
 localstatedir = @localstatedir@
-man1_MANS = @man1_MANS@
+man1_MANS =
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 oldincludedir = @oldincludedir@
@@ -14132,115 +14132,6 @@
 distclean-local:
 	test x$(srcdir) = x$(builddir) || rm -f $(ALL_MANS)
 
-$(ALL_MANS): $(mandeps)
-# Most prog.1 man pages depend on src/prog.  List the exceptions:
-@SINGLE_BINARY_FALSE@man/arch.1:      src/uname$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/install.1:   src/ginstall$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/test.1:      src/[$(EXEEXT)
-
-@SINGLE_BINARY_FALSE@man/base64.1:    src/base64$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/basename.1:  src/basename$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/cat.1:       src/cat$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/chcon.1:     src/chcon$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/chgrp.1:     src/chgrp$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/chmod.1:     src/chmod$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/chown.1:     src/chown$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/chroot.1:    src/chroot$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/cksum.1:     src/cksum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/comm.1:      src/comm$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/coreutils.1: src/coreutils$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/cp.1:        src/cp$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/csplit.1:    src/csplit$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/cut.1:       src/cut$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/date.1:      src/date$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/dd.1:        src/dd$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/df.1:        src/df$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/dir.1:       src/dir$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/dircolors.1: src/dircolors$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/dirname.1:   src/dirname$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/du.1:        src/du$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/echo.1:      src/echo$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/env.1:       src/env$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/expand.1:    src/expand$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/expr.1:      src/expr$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/factor.1:    src/factor$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/false.1:     src/false$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/fmt.1:       src/fmt$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/fold.1:      src/fold$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/groups.1:    src/groups$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/head.1:      src/head$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/hostid.1:    src/hostid$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/hostname.1:  src/hostname$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/id.1:        src/id$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/join.1:      src/join$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/kill.1:      src/kill$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/link.1:      src/link$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/ln.1:        src/ln$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/logname.1:   src/logname$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/ls.1:        src/ls$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/md5sum.1:    src/md5sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/mkdir.1:     src/mkdir$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/mkfifo.1:    src/mkfifo$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/mknod.1:     src/mknod$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/mktemp.1:    src/mktemp$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/mv.1:        src/mv$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/nice.1:      src/nice$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/nl.1:        src/nl$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/nohup.1:     src/nohup$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/nproc.1:     src/nproc$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/numfmt.1:    src/numfmt$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/od.1:        src/od$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/paste.1:     src/paste$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/pathchk.1:   src/pathchk$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/pinky.1:     src/pinky$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/pr.1:        src/pr$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/printenv.1:  src/printenv$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/printf.1:    src/printf$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/ptx.1:       src/ptx$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/pwd.1:       src/pwd$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/readlink.1:  src/readlink$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/realpath.1:  src/realpath$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/rm.1:        src/rm$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/rmdir.1:     src/rmdir$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/runcon.1:    src/runcon$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/seq.1:       src/seq$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sha1sum.1:   src/sha1sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sha224sum.1: src/sha224sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sha256sum.1: src/sha256sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sha384sum.1: src/sha384sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sha512sum.1: src/sha512sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/shred.1:     src/shred$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/shuf.1:      src/shuf$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sleep.1:     src/sleep$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sort.1:      src/sort$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/split.1:     src/split$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/stat.1:      src/stat$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/stdbuf.1:    src/stdbuf$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/stty.1:      src/stty$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sum.1:       src/sum$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/sync.1:      src/sync$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tac.1:       src/tac$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tail.1:      src/tail$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tee.1:       src/tee$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/timeout.1:   src/timeout$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/touch.1:     src/touch$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tr.1:        src/tr$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/true.1:      src/true$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/truncate.1:  src/truncate$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tsort.1:     src/tsort$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/tty.1:       src/tty$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/uname.1:     src/uname$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/unexpand.1:  src/unexpand$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/uniq.1:      src/uniq$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/unlink.1:    src/unlink$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/uptime.1:    src/uptime$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/users.1:     src/users$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/vdir.1:      src/vdir$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/wc.1:        src/wc$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/who.1:       src/who$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/whoami.1:    src/whoami$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/yes.1:       src/yes$(EXEEXT)
-
 .x.1:
 	$(AM_V_GEN)name=`echo $@ | sed 's|.*/||; s|\.1$$||'` || exit 1;	\
 	case $$name in							\
