diff --git a/console-client/Makefile b/console-client/Makefile
index 03dad1f..18dfe08 100644
--- a/console-client/Makefile
+++ b/console-client/Makefile
@@ -52,8 +52,12 @@ driver-CPPFLAGS = -D'CONSOLE_DEFPATH="$(module-dir)\0"' \
 driver-DEPS = $(..)config.make
 
 console: $(CONSOLE_SRCS:.c=.o) \
-	../libnetfs/libnetfs.a ../libfshelp/libfshelp.a \
-	../libcons/libcons.a ../libports/libports.a \
+	../libcons/libcons.a \
+	../libfshelp/libfshelp.a \
+	../libnetfs/libnetfs.a \
+	../libiohelp/libiohelp.a \
+	../libihash/libihash.a \
+	../libports/libports.a \
 	../libshouldbeinlibc/libshouldbeinlibc.a
 
 modules = vga pc_kbd generic_speaker pc_mouse current_vcs
diff --git a/daemons/Makefile b/daemons/Makefile
index db1acc7..9d32617 100644
--- a/daemons/Makefile
+++ b/daemons/Makefile
@@ -38,7 +38,8 @@ include ../Makeconf
 rc: rc.sh
 getty: getty.o ../libshouldbeinlibc/libshouldbeinlibc.a
 mail.local: lmail.o ../libshouldbeinlibc/libshouldbeinlibc.a
-console-run: console-run.o ../libfshelp/libfshelp.a ../libports/libports.a
+console-run: console-run.o ../libfshelp/libfshelp.a ../libports/libports.a \
+	../libihash/libihash.a ../libshouldbeinlibc/libshouldbeinlibc.a
 
 runttys: runttys.o
 runttys-LDLIBS = -lutil
diff --git a/devnode/Makefile b/devnode/Makefile
index 9529fa7..0964f8d 100644
--- a/devnode/Makefile
+++ b/devnode/Makefile
@@ -20,7 +20,7 @@ makemode := server
 
 SRCS = devnode.c
 LCLHDRS = util.h
-HURDLIBS = ports trivfs fshelp shouldbeinlibc
+HURDLIBS = fshelp ihash iohelp ports shouldbeinlibc trivfs
 target = devnode
 MIGSTUBS = deviceServer.o notifyServer.o
 MIGSFLAGS = -imacros $(srcdir)/mig-mutate.h
diff --git a/random/Makefile b/random/Makefile
index 0949b63..5bdc9b1 100644
--- a/random/Makefile
+++ b/random/Makefile
@@ -24,7 +24,7 @@ target = random
 SRCS = random.c gnupg-random.c gnupg-rmd160.c
 OBJS = $(SRCS:.c=.o) startup_notifyServer.o
 LCLHDRS = gnupg-random.h gnupg-rmd.h gnupg-bithelp.h random.h
-HURDLIBS = trivfs ports fshelp ihash shouldbeinlibc
+HURDLIBS = trivfs ports fshelp ihash iohelp shouldbeinlibc
 OTHERLIBS = -lpthread
 
 include ../Makeconf
diff --git a/trans/Makefile b/trans/Makefile
index ce1eae7..2fdcd38 100644
--- a/trans/Makefile
+++ b/trans/Makefile
@@ -55,22 +55,21 @@ include ../Makeconf
 
 vpath elfcore.c $(top_srcdir)/exec
 
-symlink: fsysServer.o
-ifsock: ifsockServer.o
 crash: crashServer.o crash_replyUser.o msgServer.o elfcore.o
+ifsock: ifsockServer.o
+mtab: fsUser.o
 password: passwordServer.o
-streamio: device_replyServer.o
 proxy-defpager: default_pagerServer.o default_pagerUser.o
+streamio: device_replyServer.o
+symlink: fsysServer.o
 
-proxy-defpager crash password streamio: ../libports/libports.a ../libtrivfs/libtrivfs.a ../libfshelp/libfshelp.a
+fakeroot: ../libnetfs/libnetfs.a
 fifo new-fifo: ../libpipe/libpipe.a
-fwd: ../libfshelp/libfshelp.a ../libports/libports.a
-hello-mt magic null ifsock fifo new-fifo firmlink: ../libtrivfs/libtrivfs.a ../libfshelp/libfshelp.a ../libports/libports.a ../libihash/libihash.a
-magic: ../libiohelp/libiohelp.a
-hello: ../libtrivfs/libtrivfs.a ../libfshelp/libfshelp.a ../libports/libports.a ../libihash/libihash.a
-fakeroot: ../libnetfs/libnetfs.a ../libfshelp/libfshelp.a ../libiohelp/libiohelp.a ../libports/libports.a ../libihash/libihash.a
-remap: ../libtrivfs/libtrivfs.a ../libfshelp/libfshelp.a ../libports/libports.a ../libihash/libihash.a
-mtab: ../libtrivfs/libtrivfs.a ../libfshelp/libfshelp.a ../libports/libports.a ../libihash/libihash.a fsUser.o
-$(targets): ../libshouldbeinlibc/libshouldbeinlibc.a
+crash fifo firmlink hello hello-mt ifsock magic mtab new-fifo null password proxy-defpager remap streamio: ../libtrivfs/libtrivfs.a
+$(targets): ../libfshelp/libfshelp.a \
+	../libihash/libihash.a \
+	../libiohelp/libiohelp.a \
+	../libports/libports.a \
+	../libshouldbeinlibc/libshouldbeinlibc.a
 
 $(targets): %: %.o
diff --git a/utils/Makefile b/utils/Makefile
index 955789b..603b722 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -54,12 +54,13 @@ login addauth setauth: nonsugid.o
 addauth rmauth setauth unsu: frobauth.o
 rmauth setauth unsu: frobauth-mod.o
 ps w: psout.o ../libps/libps.a ../libihash/libihash.a
-portinfo: ../libps/libps.a
+portinfo: ../libihash/libihash.a ../libps/libps.a
 
 storeinfo storecat storeread: ../libstore/libstore.a
 ftpcp ftpdir: ../libftpconn/libftpconn.a
-
-settrans: ../libfshelp/libfshelp.a ../libports/libports.a
+mount umount: ../libihash/libihash.a
+settrans: ../libfshelp/libfshelp.a ../libihash/libihash.a \
+	../libports/libports.a
 ps w ids settrans syncfs showtrans fsysopts storeinfo login vmstat portinfo \
   devprobe vminfo addauth rmauth setauth unsu ftpcp ftpdir storeread \
   storecat msgport mount umount nullauth rpctrace: \
