Index: libgpg-error-1.46/src/Makefile.am
===================================================================
--- libgpg-error-1.46.orig/src/Makefile.am
+++ libgpg-error-1.46/src/Makefile.am
@@ -67,6 +67,7 @@ lock_obj_pub = \
         syscfg/lock-obj-pub.sparc-unknown-linux-gnu.h       \
         syscfg/lock-obj-pub.sparc64-unknown-linux-gnu.h     \
 	syscfg/lock-obj-pub.x86_64-apple-darwin.h           \
+        syscfg/lock-obj-pub.x86_64-unknown-gnu.h              \
         syscfg/lock-obj-pub.x86_64-unknown-kfreebsd-gnu.h   \
         syscfg/lock-obj-pub.x86_64-unknown-linux-gnu.h      \
         syscfg/lock-obj-pub.x86_64-unknown-linux-gnux32.h   \
Index: libgpg-error-1.46/src/mkheader.c
===================================================================
--- libgpg-error-1.46.orig/src/mkheader.c
+++ libgpg-error-1.46/src/mkheader.c
@@ -108,6 +108,8 @@ canon_host_triplet (const char *triplet,
     {"x86_64-pc-linux-gnuhardened1", "x86_64-unknown-linux-gnu" },
     {"x86_64-pc-linux-gnu" },
 
+    {"x86_64-pc-gnu"},
+
     {"powerpc-unknown-linux-gnuspe", "powerpc-unknown-linux-gnu" },
 
     {"arm-unknown-linux-gnueabihf",  "arm-unknown-linux-gnueabi" },
Index: libgpg-error-1.46/src/syscfg/lock-obj-pub.x86_64-unknown-gnu.h
===================================================================
--- /dev/null
+++ libgpg-error-1.46/src/syscfg/lock-obj-pub.x86_64-unknown-gnu.h
@@ -0,0 +1,24 @@
+## lock-obj-pub.x86_64-pc-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[32];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
