diff -Naur cross-hurd/tmp/src/gcc-12.2.0/gcc/config/i386/gnu64.h gcc-12.2.0/gcc/config/i386/gnu64.h
--- cross-hurd/tmp/src/gcc-12.2.0/gcc/config/i386/gnu64.h	2022-12-10 01:04:58.039757765 -0500
+++ gcc-12.2.0/gcc/config/i386/gnu64.h	2022-12-10 21:45:27.090296782 -0500
@@ -24,9 +24,9 @@
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER32 "/lib/ld.so"
-#define GNU_USER_DYNAMIC_LINKER64 "/lib/ld-x86-64.so"
-#define GNU_USER_DYNAMIC_LINKERX32 "/lib/ld-x32.so"
+#define GNU_USER_DYNAMIC_LINKER32 "/tools/lib/ld.so"
+#define GNU_USER_DYNAMIC_LINKER64 "/tools/lib/ld-x86-64.so"
+#define GNU_USER_DYNAMIC_LINKERX32 "/tools/lib/ld-x32.so"
 
 #undef	STARTFILE_SPEC
 #if defined HAVE_LD_PIE
